File: Planets/Thear/Faction.py
class Faction:
    def __init__(self, name):
        self.name = name
        self.sub_factions = []

    def add_sub_faction(self, sub_faction):
        if sub_faction not in self.sub_factions:
            self.sub_factions.append(sub_faction)
        else:
            print(f"{sub_faction.name} already exists in {self.name}.")

    def remove_sub_faction(self, sub_faction):
        if sub_faction in self.sub_factions:
            self.sub_factions.remove(sub_faction)
        else:
            print(f"{sub_faction.name} does not exist in {self.name}.")


class FactionHierarchy:
    def __init__(self):
        self.thear = Faction("Thear")
        self.initialize_factions()

    def initialize_factions(self):
        bractalia = Faction("Bractalia")
        lochtou = Faction("Lochtou")
        kinderyarn = Faction("Kinderyarn")
        nymenada = Faction("Nymenada")
        wolk = Faction("Wolk")
        varthek = Faction("Varthek")
        jight = Faction("Jight")
        slake = Faction("Slake")
        thipse = Faction("Thipse")

        self.thear.sub_factions.extend([bractalia, lochtou, kinderyarn, nymenada, wolk, varthek, jight, slake, thipse])

    def create_faction_with_sub_factions(self, name, parent, sub_faction_names):
        faction = Faction(name)
        for sub_faction_name in sub_faction_names:
            sub_faction = Faction(sub_faction_name)
            if sub_faction not in faction.sub_factions:
                faction.add_sub_faction(sub_faction)
            else:
                print(f"{sub_faction.name} already exists in {faction.name}.")
        parent.add_sub_faction(faction)


# Example usage
faction_hierarchy = FactionHierarchy()
print("Faction hierarchy created successfully.")

File: Planets/Thear/Faction.js
// File = robjam1990/Psychosis/Gameplay/Planets/Thear/Faction.js
class Faction {
    constructor(name) {
        this.name = name;
        this.subFactions = [];
    }

    addSubFaction(subFaction) {
        if (!this.subFactions.includes(subFaction)) {
            this.subFactions.push(subFaction);
        } else {
            console.log(`${subFaction.name} already exists in ${this.name}.`);
        }
    }

    removeSubFaction(subFaction) {
        const index = this.subFactions.indexOf(subFaction);
        if (index !== -1) {
            this.subFactions.splice(index, 1);
        } else {
            console.log(`${subFaction.name} does not exist in ${this.name}.`);
        }
    }
}

class FactionHierarchy {
    constructor() {
        this.thear = new Faction("Thear");
        this.initializeFactions();
    }

    initializeFactions() {
        let bractalia = new Faction("Bractalia");
        let lochtou = new Faction("Lochtou");
        let kinderyarn = new Faction("Kinderyarn");
        let nymenada = new Faction("Nymenada");
        let wolk = new Faction("Wolk");
        let varthek = new Faction("Varthek");
        let jight = new Faction("Jight");
        let slake = new Faction("Slake");
        let thipse = new Faction("Thipse");

        this.thear.subFactions.push(bractalia, lochtou, kinderyarn, nymenada, wolk, varthek, jight, slake, thipse);
    }

    createFactionWithSubFactions(name, parent, subFactionNames) {
        let faction = new Faction(name);
        subFactionNames.forEach(subFactionName => {
            let subFaction = new Faction(subFactionName);
            if (!faction.subFactions.some(sub => sub.name === subFaction.name)) {
                faction.addSubFaction(subFaction);
            } else {
                console.log(`${subFaction.name} already exists in ${faction.name}.`);
            }
        });
        parent.addSubFaction(faction);
    }


}

// Example usage
const factionHierarchy = new FactionHierarchy();
console.log("Faction hierarchy created successfully.");
